tracer/src/Datadog.Trace/ClrProfiler/DistributedTracer.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="DistributedTracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// Used to distribute traces across multiple versions of the tracer
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class DistributedTracer
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DistributedTracer));
+
+        static DistributedTracer()
+        {
+            try
+            {
+                var parent = GetDistributedTracer();
+
+                if (parent == null)
+                {
+                    Log.Information("Building automatic tracer");
+                    Instance = new AutomaticTracer();
+                }
+                else
+                {
+                    var parentTracer = parent.DuckCast<IAutomaticTracer>();
+
+                    Log.Information("Building manual tracer, connected to {assembly}", parent.GetType().Assembly);
```

Comment:
[nit] the `parentTracer` proxy will implement `IDuckType`, so we could also do `((IDuckType)parentTracer).Type.Assembly`. Don't know that it's necessarily faster/preferable though, just a thought

Reply:
While you're correct, I prefer to leave it as is. I'm just trying to limit the number of things that could go wrong 😛